shell/client/styles/_topbar.scss

```diff
@@ -75,8 +75,7 @@ body>.topbar {
   background-color: $topbar-background-color;
   color: #ccc;
 
-  // Using 0 rather than CSS default ("auto") allows introjs to highlight elements in the topbar.
-  z-index: 0;
+  z-index: 1;
```

Comment:
Argh, stacking contexts strike again.  We need the `.topbar` and the `.main-content` to be in the same stacking context so that either can have its children stack modals atop the other.  But then within a stacking context, things are sorted in reverse DOM order, which means that since `main-content` comes after `.topbar`, its children will stack on top of `.topbar`s children.

More on stacking contexts: https://philipwalton.com/articles/what-no-one-told-you-about-z-index/

Maybe what we should be doing with `#modalDialogWithBackdrop` is something far more complicated where we transport whatever you put in that part of the template to actually live somewhere else in the DOM where we can guarantee that it lives in the appropriate stacking context.  But then we also need to do a better job of managing focus for keyboard users, since the popup will no longer naturally be the next thing in the DOM to tab to, and when you dismiss the dialog, we'd want to return focus to wherever it was before the dialog was shown ("popping the focus stack")